public/app/features/dashboard-scene/panel-edit/PanelDataPane/PanelDataTransformations/PanelDataTransformationsTab.tsx

```diff
@@ -0,0 +1,159 @@
+  tabId = 'transformations';
+  icon: IconName = 'process';
+  private _panelManager: VizPanelManager;
+
+  getTabLabel() {
+    return 'Transformations';
+  }
+
+  getItemsCount() {
+    const dataProvider = this._panelManager.state.panel.state.$data;
+    if (dataProvider instanceof SceneDataTransformer) {
+      return dataProvider.state.transformations.length;
+    } else {
+      return null;
+    }
+  }
+
+  constructor(panelManager: VizPanelManager) {
+    super({});
+
+    this._panelManager = panelManager;
+  }
+
+  get panelManager() {
+    return this._panelManager;
+  }
+}
+
+interface TransformationEditorProps {
+  sceneDataTransformer: SceneDataTransformer;
+  onChange: (transformations: DataTransformerConfig[]) => void;
+}
+
+function TransformationsEditor(props: TransformationEditorProps) {
+  const { onChange } = props;
+  const dataState = props.sceneDataTransformer.useState();
+
+  const transformations: DataTransformerConfig[] = [];
+
+  for (const t of dataState.transformations) {
+    // We can't have any CustomTransformerOperators in the panel editor. This get's rid of them and fixes the type.
+    if ('id' in t) {
+      transformations.push(t);
+    }
+  }
+
+  const transformationEditorRows = transformations.map((t, i) => ({ id: `${i} - ${t.id}`, transformation: t }));
+
+  return (
+    <DragDropContext onDragEnd={() => {}}>
+      <Droppable droppableId="transformations-list" direction="vertical">
+        {(provided) => {
+          return (
+            <div ref={provided.innerRef} {...provided.droppableProps}>
+              <TransformationOperationRows
+                onChange={(index, transformation) => {
+                  const newTransformations = transformations.slice();
+                  newTransformations[index] = transformation;
+                  onChange(newTransformations);
+                }}
+                onRemove={(index) => {
+                  const newTransformations = transformations.slice();
+                  newTransformations.splice(index);
+                  onChange(newTransformations);
+                }}
+                configs={transformationEditorRows}
+                data={{
+                  series: dataState.data?.series || [],
+                  annotations: dataState.data?.annotations || [],
+                }}
+              ></TransformationOperationRows>
+              {provided.placeholder}
+            </div>
+          );
+        }}
+      </Droppable>
+    </DragDropContext>
+  );
+}
+
+export function PanelDataTransformationsTabRendered({ model }: SceneComponentProps<PanelDataTransformationsTab>) {
+  const styles = useStyles2(getStyles);
+  const panelManagerState = model.panelManager.useState();
+  const panelState = panelManagerState.panel.useState();
+
+  if (!(panelState.$data instanceof SceneDataTransformer)) {
+    return;
+  }
+
+  const dataState = panelState.$data.useState();
+
+  return (
+    <CustomScrollbar autoHeightMin="100%">
+      <Container>
```

Comment:
can we move these CustomScrollbar + Container to PanelDataPaneRendered (so shared between all tabs). 

Then we can simplify do:

```ts
if (dataState.transformations.length < 1) {
  return <EmptyTransformationsMessage onShowPicker={() => {}} />
}
```

I really struggle reading react markup with long multiline co ternary operators :)